File: Backend/Repositories/Python/sms.py
import sqlite3

# Initialize database
def init_db():
    conn = sqlite3.connect('students.db')
    cursor = conn.cursor()
    cursor.execute('''
        CREATE TABLE IF NOT EXISTS students (
            id INTEGER PRIMARY KEY AUTOINCREMENT,
            name TEXT NOT NULL,
            age INTEGER,
            grade TEXT
        )
    ''')
    conn.commit()
    conn.close()

# Add new student
def add_student(name, age, grade):
    conn = sqlite3.connect('students.db')
    cursor = conn.cursor()
    cursor.execute('INSERT INTO students (name, age, grade) VALUES (?, ?, ?)', (name, age, grade))
    conn.commit()
    conn.close()
    print(f"Student {name} added successfully.")

# View all students
def view_students():
    conn = sqlite3.connect('students.db')
    cursor = conn.cursor()
    cursor.execute('SELECT * FROM students')
    students = cursor.fetchall()
    conn.close()
    if students:
        print("\nID | Name       | Age | Grade")
        print("-------------------------------")
        for student in students:
            print(f"{student[0]:<3} | {student[1]:<10} | {student[2]:<3} | {student[3]}")
    else:
        print("No students found.")

# Update student
def update_student(student_id, name, age, grade):
    conn = sqlite3.connect('students.db')
    cursor = conn.cursor()
    cursor.execute('UPDATE students SET name = ?, age = ?, grade = ? WHERE id = ?', (name, age, grade, student_id))
    conn.commit()
    conn.close()
    print(f"Student ID {student_id} updated successfully.")

# Delete student
def delete_student(student_id):
    conn = sqlite3.connect('students.db')
    cursor = conn.cursor()
    cursor.execute('DELETE FROM students WHERE id = ?', (student_id,))
    conn.commit()
    conn.close()
    print(f"Student ID {student_id} deleted successfully.")

# Main menu
def main():
    init_db()
    while True:
        print("\n--- Student Management System ---")
        print("1. Add Student")
        print("2. View Students")
        print("3. Update Student")
        print("4. Delete Student")
        print("5. Exit")
        choice = input("Enter your choice: ")

        if choice == '1':
            name = input("Enter student name: ")
            age = int(input("Enter student age: "))
            grade = input("Enter student grade: ")
            add_student(name, age, grade)

        elif choice == '2':
            view_students()

        elif choice == '3':
            student_id = int(input("Enter student ID to update: "))
            name = input("Enter new name: ")
            age = int(input("Enter new age: "))
            grade = input("Enter new grade: ")
            update_student(student_id, name, age, grade)

        elif choice == '4':
            student_id = int(input("Enter student ID to delete: "))
            delete_student(student_id)

        elif choice == '5':
            print("Exiting... Goodbye!")
            break

        else:
            print("Invalid choice. Please try again.")

if __name__ == '__main__':
    main()

File: Backend/Repositories/Java/sms.java
import java.util.*;
import java.io.*;

class Student {
    private int id;
    private String name;
    private String course;
    private double gpa;

    public Student(int id, String name, String course, double gpa) {
        this.id = id;
        this.name = name;
        this.course = course;
        this.gpa = gpa;
    }

    public int getId() { return id; }
    public String getName() { return name; }
    public String getCourse() { return course; }
    public double getGpa() { return gpa; }

    public void display() {
        System.out.println("ID: " + id + ", Name: " + name + ", Course: " + course + ", GPA: " + gpa);
    }
}

class StudentManager {
    private List<Student> students = new ArrayList<>();

    public void addStudent(int id, String name, String course, double gpa) {
        students.add(new Student(id, name, course, gpa));
        System.out.println("Student added successfully!\n");
    }

    public void displayAll() {
        if (students.isEmpty()) {
            System.out.println("No students in the system.\n");
            return;
        }
        System.out.println("--- Student List ---");
        for (Student s : students) {
            s.display();
        }
    }

    public void searchStudent(int id) {
        for (Student s : students) {
            if (s.getId() == id) {
                System.out.println("Student found:");
                s.display();
                return;
            }
        }
        System.out.println("Student not found.\n");
    }

    public void deleteStudent(int id) {
        Iterator<Student> iterator = students.iterator();
        while (iterator.hasNext()) {
            if (iterator.next().getId() == id) {
                iterator.remove();
                System.out.println("Student deleted successfully.\n");
                return;
            }
        }
        System.out.println("Student not found.\n");
    }

    public void saveToFile(String filename) {
        try (BufferedWriter writer = new BufferedWriter(new FileWriter(filename))) {
            for (Student s : students) {
                writer.write(s.getId() + "," + s.getName() + "," + s.getCourse() + "," + s.getGpa());
                writer.newLine();
            }
            System.out.println("Student data saved to file.\n");
        } catch (IOException e) {
            System.out.println("Error saving to file: " + e.getMessage());
        }
    }

    public void loadFromFile(String filename) {
        try (BufferedReader reader = new BufferedReader(new FileReader(filename))) {
            String line;
            students.clear();
            while ((line = reader.readLine()) != null) {
                String[] parts = line.split(",");
                if (parts.length == 4) {
                    int id = Integer.parseInt(parts[0]);
                    String name = parts[1];
                    String course = parts[2];
                    double gpa = Double.parseDouble(parts[3]);
                    students.add(new Student(id, name, course, gpa));
                }
            }
            System.out.println("Student data loaded from file.\n");
        } catch (IOException | NumberFormatException e) {
            System.out.println("Error loading from file: " + e.getMessage());
        }
    }
}

public class StudentManagementSystem {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        StudentManager manager = new StudentManager();
        int choice;

        do {
            System.out.println("\n--- Student Management System ---");
            System.out.println("1. Add Student\n2. Display All Students\n3. Search Student\n4. Delete Student\n5. Save to File\n6. Load from File\n0. Exit");
            System.out.print("Enter your choice: ");
            choice = scanner.nextInt();

            switch (choice) {
                case 1:
                    System.out.print("Enter ID: ");
                    int id = scanner.nextInt();
                    scanner.nextLine();
                    System.out.print("Enter Name: ");
                    String name = scanner.nextLine();
                    System.out.print("Enter Course: ");
                    String course = scanner.nextLine();
                    System.out.print("Enter GPA: ");
                    double gpa = scanner.nextDouble();
                    manager.addStudent(id, name, course, gpa);
                    break;
                case 2:
                    manager.displayAll();
                    break;
                case 3:
                    System.out.print("Enter ID to search: ");
                    int searchId = scanner.nextInt();
                    manager.searchStudent(searchId);
                    break;
                case 4:
                    System.out.print("Enter ID to delete: ");
                    int deleteId = scanner.nextInt();
                    manager.deleteStudent(deleteId);
                    break;
                case 5:
                    System.out.print("Enter filename to save: ");
                    scanner.nextLine();
                    String saveFile = scanner.nextLine();
                    manager.saveToFile(saveFile);
                    break;
                case 6:
                    System.out.print("Enter filename to load: ");
                    scanner.nextLine();
                    String loadFile = scanner.nextLine();
                    manager.loadFromFile(loadFile);
                    break;
                case 0:
                    System.out.println("Exiting Student Management System. Goodbye!");
                    break;
                default:
                    System.out.println("Invalid choice. Try again.\n");
            }
        } while (choice != 0);
        scanner.close();
    }
}
